src/components/manager.ts

```diff
@@ -84,10 +84,9 @@ export class Manager {
      */
     private readonly cachedContent = Object.create(null) as Content
 
-    private readonly localRootFiles = Object.create(null) as { [key: string]: string | undefined }
-    private readonly rootFilesLanguageIds = Object.create(null) as { [key: string]: string | undefined }
-    // Store one root file for each workspace.
-    private readonly rootFiles = Object.create(null) as { [key: string]: RootFileType | undefined }
+    private _localRootFile: string | undefined
+    private _rootFilesLanguageId: string | undefined
```

Comment:
Singular form `_rootFileLanguageId` would be better.

Reply:
For sure!